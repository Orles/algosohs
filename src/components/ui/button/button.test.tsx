import { render, screen, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer'
import { Button } from "./button";

describe('Тест кнопки', () => {
    it('кнопка с текстом', () => {
        const button = renderer.create(<Button text="text" />)
        expect(button).toMatchSnapshot()
    })
    it('кнопка без текста', () => {
        const button = renderer.create(<Button />)
        expect(button).toMatchSnapshot()
    })
    it('заблокированная кнопка', () => {
        const button = renderer.create(<Button disabled={true} />)
        expect(button).toMatchSnapshot()
    })
    it('кнопка с индикацией загрузки', () => {
        const button = renderer.create(<Button isLoader={true} />)
        expect(button).toMatchSnapshot()
    })
    it('Корректность вызова колбека при клике на кнопку', () => {
        const click = jest.fn();
        render(<Button text="button" onClick={click} />)
        const button = screen.getByText("button");
        fireEvent.click(button)
        expect(click).toHaveBeenCalled()
    })
})