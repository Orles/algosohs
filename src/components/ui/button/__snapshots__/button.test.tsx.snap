// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест кнопки заблокированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест кнопки кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест кнопки кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тест кнопки кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
